_source/_posts/2022-08-24-state-of-project-loom.md

```diff
@@ -0,0 +1,261 @@
+---
+layout: blog_post
+title: "What the Heck Is Java Project Loom?"
+author: deepu-sasidharan
+by: advocate
+communities: [java]
+description: "What the heck is Java Project Loom and what does it mean for average Java developers?"
+tags: [java, concurrency, project-loom]
+tweets:
+  - "What the heck is #Java Project Loom and what does it mean for average Java developers?"
+  - "What is the hottest new thing in #Java? Learn all about Virtual threads and Project Loom"
+image:
+type: awareness
+---
+
+Java has good multi-threading and concurrency capabilities from early on in its evolution and could effectively utilize multi-threaded and multi-core CPUs. JDK 1.1 had basic support for platform threads (or Operating System (OS) threads), and JDK 1.5 had more utilities and updates to improve concurrency and multi-threading. JDK 8 brought asynchronous programming support and more concurrency improvements. While things improved over multiple versions, there was nothing groundbreaking apart from support for concurrency and multi-threading using OS threads for the last three decades.
+
+Though the concurrency model in Java was powerful and flexible as a feature, it was not the easiest to use, and the developer experience was not that great. This was primarily due to the shared state concurrency model used by default. One has to resort to synchronizing threads to avoid issues like data races and thread blocking. I wrote more about Java concurrency in my [Concurrency in modern programming languages: Java](https://deepu.tech/concurrency-in-modern-languages-java/) post.
```

Comment:
```suggestion
Though the concurrency model in Java is powerful and flexible as a feature, it was not the easiest to use, and the developer experience hasn't been great. This is primarily due to the shared state concurrency model used by default. One has to resort to synchronizing threads to avoid issues like data races and thread blocking. I wrote more about Java concurrency in my [Concurrency in modern programming languages: Java](https://deepu.tech/concurrency-in-modern-languages-java/) post.
```